apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -869,84 +869,45 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
-      :xandra,
-      &do_check_device_deletion_in_progress(&1, keyspace_name, device_id)
-    )
-  end
-
-  defp do_check_device_deletion_in_progress(conn, realm_name, device_id) do
-    statement = """
-    SELECT *
-    FROM #{realm_name}.deletion_in_progress
-    WHERE device_id = :device_id
-    """
-
-    opts = [
-      consistency: Consistency.device_info(:read),
-      uuid_format: :binary
-    ]
-
-    with {:ok, prepared} <- Xandra.prepare(conn, statement),
-         {:ok, %Xandra.Page{} = page} <-
-           Xandra.execute(conn, prepared, %{"device_id" => device_id}, opts) do
-      result_not_empty? = not Enum.empty?(page)
-      {:ok, result_not_empty?}
-    else
-      {:error, %Xandra.Error{} = error} ->
-        _ =
-          Logger.warning(
-            "Database error while checking device deletion in progress: #{Exception.message(error)}"
-          )
+    keyspace = Realm.keyspace_name(realm_name)
 
-        {:error, :database_error}
+    query =
+      from d in DeletionInProgress,
+        prefix: ^keyspace,
+        where: d.device_id == ^device_id
 
-      {:error, %Xandra.ConnectionError{} = error} ->
-        _ =
-          Logger.warning(
-            "Database connection error while checking device deletion in progress: #{Exception.message(error)}"
-          )
+    consistency = Consistency.device_info(:read)
 
-        {:error, :database_connection_error}
+    with {:ok, result} <- Repo.fetch_all(query, consistency: consistency) do
+      {:ok, Enum.any?(result)}
     end
   end
 
   def retrieve_realms! do
-    statement = """
-    SELECT *
-    FROM #{CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())}.realms
-    """
-
-    realms =
-      Xandra.Cluster.run(
-        :xandra,
-        &Xandra.execute!(&1, statement, %{}, consistency: Consistency.domain_model(:read))
-      )
+    keyspace_name = Realm.astarte_keyspace_name()
+
+    query =
+      from Realm,
+        prefix: ^keyspace_name
+
+    consistency = Consistency.domain_model(:read)
 
-    Enum.to_list(realms)
+    with {:ok, realms} <- Repo.fetch_all(query, consistency: consistency) do
+      realms
+    end
   end
 
   def retrieve_devices_waiting_to_start_deletion!(realm_name) do
     keyspace_name = Realm.keyspace_name(realm_name)
 
-    Xandra.Cluster.run(
-      :xandra,
-      &do_retrieve_devices_waiting_to_start_deletion!(&1, keyspace_name)
-    )
-  end
+    query =
+      from DeletionInProgress,
+        prefix: ^keyspace_name
 
-  defp do_retrieve_devices_waiting_to_start_deletion!(conn, realm_name) do
-    statement = """
-    SELECT *
-    FROM #{realm_name}.deletion_in_progress
-    """
+    consistency = Consistency.domain_model(:read)
 
-    Xandra.execute!(conn, statement, %{},
-      consistency: Consistency.device_info(:read),
-      uuid_format: :binary
-    )
-    |> Enum.to_list()
+    with {:ok, devices} <- Repo.fetch_all(query, consistency: consistency) do
+      devices
+    end
```

Comment:
This was previously a bang

```suggestion
    Repo.all(query, consistency: consistency)
```

nit: also tbf you don't need to specify a custom query for each repo operation, especially `select *` are quite ergonomic

```suggestion
    Repo.all(DeletionInProgress, prefix: keyspace_name, consistency: consistency)
```